src/pages/index.js

```diff
@@ -1,95 +1,245 @@
+// ! ||--------------------------------------------------------------------------------||
+// ! ||                                    Imports                                     ||
+// ! ||--------------------------------------------------------------------------------||
+
 import "./index.css";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
 import ModalWithForm from "../components/ModalWithForm.js";
+import ModalWithFormSubmit from "../components/ModalWithFormSubmit.js";
 import ModalWithImage from "../components/ModalWithImage.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api";
 import {
-  initialCards,
   validationSettings,
   cardListSelector,
   profileForm,
   addCardFormElement,
   profileEditButton,
   profileName,
   profileDescription,
+  avatarImage,
   addNewCardButton,
   profileNameInput,
+  avatarButton,
   profileDescriptionInput,
 } from "../utils/constants.js";
 
-const profileFormValidator = new FormValidator(validationSettings, profileForm);
-
-profileFormValidator.enableValidation();
-
-const addCardFormValidator = new FormValidator(
-  validationSettings,
-  addCardFormElement
-);
-
 // ! ||--------------------------------------------------------------------------------||
-// ! ||                                    Elements                                    ||
+// ! ||                        Initialization and Configuration                        ||
 // ! ||--------------------------------------------------------------------------------||
 
-const addCardModal = new ModalWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
-);
-
 const cardsList = new Section(
   {
-    items: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
+    renderer: (card) => {
+      const cardElement = createCard(card);
       cardsList.addItem(cardElement);
     },
   },
   cardListSelector
 );
+const userInfo = new UserInfo({
+  profileName: profileName,
+  profileDescription: profileDescription,
+  avatarImage: avatarImage,
+});
 
-const imageModal = new ModalWithImage("#preview-image-modal");
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+});
+
+const profileFormValidator = new FormValidator(validationSettings, profileForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```